pkg/scalers/kafka_scaler.go

```diff
@@ -50,15 +51,13 @@ const (
 	earliest offsetResetPolicy = "earliest"
 )
 
-type kafkaAuthMode string
+type kafkaSaslType string
 
+// supported SASL types
 const (
-	kafkaAuthModeForNone            kafkaAuthMode = "none"
-	kafkaAuthModeForSaslPlaintext   kafkaAuthMode = "sasl_plaintext"
-	kafkaAuthModeForSaslScramSha256 kafkaAuthMode = "sasl_scram_sha256"
-	kafkaAuthModeForSaslScramSha512 kafkaAuthMode = "sasl_scram_sha512"
-	kafkaAuthModeForSaslSSL         kafkaAuthMode = "sasl_ssl"
-	kafkaAuthModeForSaslSSLPlain    kafkaAuthMode = "sasl_ssl_plain"
+	KafkaSASLTypePlaintext   kafkaSaslType = "plaintext"
+	KafkaSASLTypeSCRAMSHA256 kafkaSaslType = "scram_sha256"
+	KafkaSASLTypeSCRAMSHA512 kafkaSaslType = "scram_sha512"
```

Comment:
what about a SASL type "none" instead of using another field like `enableSASL` to describe that SASL is enabled or not?

Reply:
Is this approach typically used in Kafka ecosystem? I wanted make the config as simple as possible for users, the `enableSASL` is validated in the code only, it is not exposed to the user. And I don't see added value for `none`, if user don't want to use SASL, he just don't specify this value. But if this (addind `none`) is something that is expected from Kafka users POV, I am happy to do that.

Here is the relevant doc on how to configure SASL and TLS:
https://github.com/kedacore/keda-docs/pull/242/files#diff-0d86a63b316b8e7395db2a6ddc6d56dfR40-R49

Reply:
Totally agree what I meant is that ...
The user has to specify the `sasl` field in the YAML or ... nothing if he doesn't want SASL.
At level code why not translating the absence of `sasl` in a mode = "none" instead of a flag enabled = false? Just to avoid the flag. I don't mean that the user should put `sals: none` in the YAML.

Reply:
Gotcha, gonna change that. Thx.

@ppatierno  do you have any thoughts on https://github.com/kedacore/keda-docs/pull/242/files#r483445597 ? Thanks!